jerry-core/ecma/builtin-objects/ecma-builtin-array.c

```diff
@@ -499,7 +500,29 @@ ecma_builtin_array_dispatch_construct (const ecma_value_t *arguments_list_p, /**
 {
   JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
 
-  return ecma_op_create_array_object (arguments_list_p, arguments_list_len, true);
+  ecma_value_t result = ecma_op_create_array_object (arguments_list_p, arguments_list_len, true);
+
+  if (ECMA_IS_VALUE_ERROR (result))
+  {
+    return result;
+  }
+#if ENABLED (JERRY_ES2015)
+  if (JERRY_CONTEXT (current_new_target) != NULL)
```

Comment:
In `ecma_builtin_array_dispatch_call` return directly with the result of the `ecma_op_create_array_object `, so this check is not needed.

Reply:
This request is not resolved.

Reply:
At this point we can assert that `JERRY_CONTEXT (current_new_target) != NULL`. So no if check is needed. The request was about to change  `ecma_builtin_array_dispatch_call ` to return diretly with `return ecma_op_create_array_object (arguments_list_p, arguments_list_len, true);`